src/Avalonia.Base/Platform/IBitmapWithPixelReadAccessImpl.cs

```diff
@@ -0,0 +1,7 @@
+namespace Avalonia.Platform;
+
+public interface IReadableBitmapImpl
```

Comment:
File name and interface names are not matching.

Reply:
Also doesn't `ILockedFramebuffer Lock()` provide write access to the bitmap as well?

Reply:
It does, but it's `Impl`-suffixed interface that's not a public API.

In general WriteableBitmap should have `AddDirtyRect` API to notify Avalonia that something got changed.